### ISSUE TITLE
rsyslog does not forward received logs

### ISSUE THREAD

**Issue by shaonianlang-ooo** [rsyslog/rsyslog#5416]
### Expected behavior
Rsyslog receives external logs and forwards them
### Actual behavior
Rsyslog receives external logs and stores them locally according to the date. A problem occurs during the forwarding process: the number of messages received is 1, which is correct and stored in a new log file, but Rsyslog does not read the log, as if it does not detect the existence of this file. Rsyslog needs to be restarted to read and forward. Why is this?
### Steps to reproduce the behavior
This situation will occur as long as new logs are generated, the number of logs is small, and no new logs are generated for a long time after they are received once. The system will read them only after restarting.
### Environment
- rsyslog version: 
- platform: 
- for configuration questions/issues, include rsyslog.conf and included config files
rsyslogd 8.32.0
Ubuntu 18.04.6 LTS (GNU/Linux 4.15.0-213-generic x86_64)

Receiving log configuration：
$template CustomMsgFormat,"%fromhost-ip% %msg%\n"
$template CustomLogPath,"/var/spool/rsyslog/%fromhost-ip%/%$YEAR%-%$MONTH%-%$DAY%.log"

*.* ?CustomLogPath;CustomMsgFormat

Forwarding log configuration：
input(type="imfile"
      File="/var/spool/rsyslog/*/*"
      Tag="localfile"
      PersistStateInterval="5"
      ruleset="sendOtherToSyslogServerCloud")

$template CustomMsgFormatOther," %hostname% %msg%\n"

ruleset(name="sendOtherToSyslogServerCloud") {
         if ($!metadata!filename contains "127.0.0.1") then {
        stop  
    }
    action(type="omfwd"
           Target="192.168.31.2"
           Port="514"
           Protocol="tcp"
           template="CustomMsgFormatOther"
           KeepAlive="on"
           action.resumeRetryCount="-1"
           queue.type="LinkedList"
           queue.filename="remotequeue"
           queue.size="50000"
           queue.maxdiskspace="100g"
           queue.saveonshutdown="on"
           queue.timeoutenqueue="0"
           queue.timeoutshutdown="60000"
           queue.discardMark="0"
           queue.highWatermark="45000"
           queue.lowWatermark="35000"
           StreamDriver="gtls"
           StreamDriverMode="1"
           StreamDriverPermittedPeers="server.com"
           StreamDriverAuthMode="x509/name"
    )
}

**Comment by rgerhards** [comment]
Apologies for the late follow-up. This is probably no longer relevant for the original reporter, but I am adding the answer for people who find this issue through search. We do not lightly close old issues just because of age; we are working through the backlog and closing them when the old thread has a clear answer.

The configuration pattern here is the same core problem as in similar relay questions: rsyslog receives a message, writes it to a file, and then `imfile` is expected to discover and re-read rsyslog's own output file for forwarding. That is not the recommended or well-defined relay design. It can run into file discovery timing, ordering, duplication, and feedback-loop issues, especially with dynamic paths and newly-created files.

If the incoming message must be both stored and forwarded, put the local `omfile` action and the remote forwarding action in the same ruleset that handles the original input. Use an action queue on the forwarding action for buffering remote outages. If `imfile` is used for external files created by another application, then use a current rsyslog version and configure the wildcard/polling/metadata behavior explicitly; but for rsyslog-produced relay files, forwarding directly from the original message is the right answer. Closing this as answered/configuration guidance.
